### DISCUSSION TITLE
How to convert a html page back to qmd?

### DISCUSSION THREAD

**Discussion by dgdi** [quarto-dev/quarto-cli#6661] [Q&A] [answered]
### Description

Hello,

I recently discovered a few online books published via Quarto that I'm keen on using for teaching. One particular book, [Python for Data Analysis](https://wesmckinney.com/book/), offers the code notebooks on [GitHub](https://github.com/wesm/pydata-book/tree/3rd-edition), but not the original qmd. 
Creating slides can be quite time-consuming, so I'm searching for a tool that could assist in the conversion of the html pages back into a qmd format, similar to the original input. This would speed up significantly the slide preparation.
Does anyone have suggestions or solutions for this?

**Comment by cscheid** [comment]
> offers the code notebooks on [GitHub](https://github.com/wesm/pydata-book/tree/3rd-edition), but not the original qmd.

Quarto can render from .ipynb files directly; those are possibly the actual source of the book.

**Comment by dgdi** [comment]
Hi @cscheid, thanks for the quick reply!
Unfortunately the available notebooks only store the code cells (of the original qmd, I believe), without the markdown cells that make up the most part of the content.

**Reply by mcanouil** [reply]
Note that your question is kind of ethically speaking not a great one.
If the authors do not want to share the source, it is for a reason, providing a way to go against that feels not right.
You should reach out to the authors.

Also, when using others' work, please do watch for license and deal with them appropriately.

**Reply by cscheid** [reply]
> Unfortunately the available notebooks only store the code cells (of the original qmd, I believe), without the markdown cells that make up the most part of the content.

I apologize, I should have checked the notebook files. In that case, I think @mcanouil's advice is appropriate here; the author probably doesn't mean for you to use that content, and the LICENSE file in the repository does not include the textual content of the book.

**Reply by dgdi** [reply]
Hi @mcanouil, thanks for your reply.  I see your point and I have reached out to the author who told me the publisher would not allow the release of the original notebooks. 

That said, I thought of your point before starting the discussion but I am still not convinced by it.  

As a teacher who will use the book in the course, I am now expanding and rewriting (some parts of) the book content in a slide format for classroom presentation. This is standard in academia and I do not think this raises any legal/ethical issues. 
Now what I am searching for is a tool to automate and speed up this process, so that I do not have to start from scratch.  
Why should that be a problem? because an intermediate output ends up being close to the original? this does not sound right to me but, clearly, I might be wrong. 

@mcanouil I am happy to hear your view (and change my mind/close this discussion, if needed) now that I have provided a bit more context.

**Reply by cscheid** [reply, marked as answer]
Without having to resolve the ethical concerns of your current situation, the answer to your question is that quarto doesn't have a mechanism for converting from the target format to the source format.

**Comment by dgdi** [comment]
@cscheid sorry for the previous messages mixing up, I forgot to refresh before replying. Many thanks for the info, and thanks for your work, Quarto is an impressive tool! even despite the lack of this "essential" feature :- )

**Reply by ericjmorey** [reply]
For others, like myself, who have legitimate use for this feature such as converting a document licensed under a permissive license such as the [Creative Commons licenses](https://creativecommons.org/share-your-work/cclicenses/) that allow for derivatives or uses that are covered under the fair use protections or even for converting your own original documents, look into using [Pandoc](https://pandoc.org/) directly.

**Comment by carsen-stringer** [comment]
Asking here as I think it's a similar question. Can you put html files into your quarto project and have them as links from quarto pages (ideally with the github-pages hosting approach)? I have html files that I've written myself that I didn't want to have to convert, but I can if necessary. Apologies if this is answered else, thanks for the great tool!

**Reply by mcanouil** [reply]
It's actually a different question, could you open a new discussion? Thank you.
